test/goose/specs_test.clj

```diff
@@ -10,14 +10,24 @@
     [goose.test-utils :as tu]
     [goose.worker :as w]
 
-    [clojure.test :refer [deftest is are]])
+    [clojure.test :refer [deftest is are]]
+    [tech.v3.dataset :as ds])
   (:import
     (clojure.lang ExceptionInfo)
     (java.time Instant)))
 
 (defn single-arity-fn [_] "dummy")
 (def now (Instant/now))
 
+;;; A tech.v3.DS was reported to have mis-matching types post
+;;; de/serialization. This ugliness is the ONLY way to reproduce this bug.
+(def tech-v3-dataset-sample
```

Comment:
suggestion: this test is very opaque, hard to understand what its really doing.
maybe explain what the dataset is? And what is the map `{:1 1 :2 2 :3 3 :4 4 :5 5 :6 6 :7 7 :8 8}` actually doing?

Reply:
It might reproduce the bug, but as a test it needs to have clear info on what is happening.

Reply:
Added comments to explain the test.